src/olympia/lib/git.py

```diff
@@ -261,6 +265,11 @@ def git_repository(self):
 
         return git_repository
 
+    def delete(self):
+        if not self.is_extracted:
```

Comment:
This shouldn't happen, right ? Maybe we should log it

Reply:
It should not, unless we (developers) call it before the repo is initialized. I was thinking about raising an error but logging might be enough.